service/controller/v11/ebs/filter.go

```diff
@@ -0,0 +1,52 @@
+package ebs
+
+import (
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
+
+	"github.com/giantswarm/aws-operator/service/controller/v11/key"
+)
+
+const (
+	cloudProviderPersistentVolumeTagKey = "kubernetes.io/created-for/pv/name"
+	nameTagKey                          = "Name"
+)
+
+func IsFiltered(vol *ec2.Volume, filterFuncs []func(t *ec2.Tag) bool) bool {
+	for _, f := range filterFuncs {
+		for _, t := range vol.Tags {
+			if f(t) {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
+func NewDockerVolumeFilter(customObject v1alpha1.AWSConfig) func(t *ec2.Tag) bool {
```

Comment:
I think those functions could just take `[]Volume` and return `[]Volume`. To avoid `ListVolumes` `filterFunc` argument.

Reply:
This would be filtering by result and not filtering by property. I would discourage to do it this way because once the data structure changes you have to adapt filters and such weird things. I think this is a right approach.

Reply:
OK, but if the data structure change (which I think we're talking about `Volume` here), then you still will have to rename functions.

Reply:
I do not see this. Why do you think so? The filter functions are not concerned with our custom `Volume` type. They just act on a general EC2 tag.

Reply:
I got lost in this conversation. Let's not spend more energy here.

Reply:
Ack. <3